[react-native] Upgrade to 0.59.8

sjchmiela opened the pull request:
# Why 
Pull in latest changes from `react-native`.

# How
- created `sdk-33-candidate` branch by checking out `sdk-33`, running `git pull upstream 0.59-stable --rebase` and creating this new branch
- changed all `react-native` versions in all `package.json`s in the monorepo to `0.59.8` so it matches `react-native-lab/react-native/package.json#version`
- installed pods after regenerating files
- ran `yarn` in `home` and then `et publish-dev-expo-home`
- ran `gulp android-update-rn` in `tools` to update `android/React{Android,Common}`
- fixed nullability specifier (the only non-automatic change in this PR) [`9743ed46`](https://github.com/expo/expo/pull/4170/commits/9743ed46ad52a88013c81571f8d1afe7e9c5378b)

# Test plan

Home looks and works ok on iOS and Android, NCL.ReactNativeCore screen too.

sjchmiela committed:
[react-native-lab] Upgrade react-native in react-native-lab to sdk-33-candidate, i. e. sdk-33 rebased onto 0.59-stable

sjchmiela committed:
[react-native] Upgrade react-native to proper version in all package.json in the repo

sjchmiela committed:
[ios] Install pods after regenerating files

sjchmiela committed:
[home] Update dev-home-config.json with newly built Home

sjchmiela committed:
[android] Update ReactAndroid by running android-update-rn Gulp task in tools

sjchmiela committed:
[ios] Fix nullability specification for exception handler

ide commented:
CI is failing with errors like:
```
❌  duplicate symbol _OBJC_IVAR_$_RNGestureHandler._shouldCancelWhenOutside in
> libExpoKit.a(RNGestureHandler-8021E88D3B234114.o)
> libExpoKit.a(RNGestureHandler-8021E88D3B234114.o)
```

Other than that, this looks OK.

sjchmiela commented:
That error originates from `master`, introduced in https://github.com/expo/expo/commit/7f85c5775fcf401dabfcf147e24e538d7f8225b8, fixed in https://github.com/expo/expo/commit/27639e09664e5fdcc3ea302522e81cd4207cf7c3. Will rebase in a sec.
